### PR TITLE
SONAR-8332 fix broken thread-safety of UUID generation

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [logging](https://github.com/SonarSource/sonarqube/blob/f90e7746dbee4bcbd55a4f470d6292ad19955dfb/server/sonar-process/src/main/java/org/sonar/process/logging): Add a 'package-info.java' file to document the 'logging' package [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [logging](https://github.com/SonarSource/sonarqube/blob/0fea519122f70837cf3cfbb11c0039cf4c798206/server/sonar-process/src/main/java/org/sonar/process/logging): Add a 'package-info.java' file to document the 'logging' package [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

File: sonar-core/src/test/java/org/sonar/core/util/UuidGeneratorImplTest.java [hunk 1]
```diff
@@ -50,6 +53,33 @@ public void generate_returns_unique_values_without_common_initial_letter_given_m
   }
 
   @Test
+  public void generate_concurrent_test() throws InterruptedException {
```
Comment (simonbrandhof) [inline sonar-core/src/test/java/org/sonar/core/util/UuidGeneratorImplTest.java]: As a safeguard I recommend to add a JUnit timeout rule:
```
@Rule
public TestRule timeout = new DisableOnDebug(Timeout.seconds(60));
```

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [logging](https://github.com/SonarSource/sonarqube/blob/6ae077f1da40281eca98c861c2a860829db0ad2c/server/sonar-process/src/main/java/org/sonar/process/logging): Add a 'package-info.java' file to document the 'logging' package [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [logging](https://github.com/SonarSource/sonarqube/blob/aea44b9d8f223e201b5a4a2f0d85206880e76de5/server/sonar-process/src/main/java/org/sonar/process/logging): Add a 'package-info.java' file to document the 'logging' package [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

**Pull request merged**